### ISSUE TITLE
Testing on Custom Images

### ISSUE THREAD

**Issue by Jnasic** [elvisyjlin/AttGAN-PyTorch#10]
Hello, 
Thank you for your great implementation. Can you kindly tell me how can I use the pretrained model to test on my own images. 
Thanks

**Comment by elvisyjlin** [comment]
Hi @Jnasic,

It's a great idea. But my code supports testing on only CelebA and CelebA-HQ at the moment.

If you'd like to do it yourself, what you need to do is to make __a custom dataset class__ and replace `test_dataset` with it in [this line](https://github.com/elvisyjlin/AttGAN-PyTorch/blob/master/test.py#L67). The custom dataset is similar to `CelebA` dataset (see [here](https://github.com/elvisyjlin/AttGAN-PyTorch/blob/master/data.py#L16)). There is no need to read the CelebA files but __paths of your testing image__. And you need appropriate [transform functions](https://github.com/elvisyjlin/AttGAN-PyTorch/blob/master/data.py#L35) when loading images, which are resizing them to `args.img_size` and normalizing them to [-1, 1].

I'll try to make a custom image loader if I have time this weekend. If you are not in a hurry, you can just wait for a couple of days.

**Comment by elvisyjlin** [comment]
I forgot to mention that you will also need to build __an attribute list__ for your custom images like the `./data/list_attr_celeba.txt`. AttGAN encodes an image (somehow disentangles its characteristics) and then decodes it with a set of given attributes. So the model needs to know what specific attributes the image has as well as all the target attributes that you want to change it into.

So you probably have to
1. Create a file like `./data/list_attr_celeba.txt` containing the ids and attributes.
2. Name your testing images as `000000.jpg`, `000001.jpg`, etc.
3. Read them with a dataset like `CelebA` but without cropping images into 170x170.

It requires a lot of work though.

**Comment by elvisyjlin** [comment]
Hi, I've added custom image dataset in 2f8cb3e. You can test on your custom images after preparing images and the attribute list in `./data`. Please read the instruction [here](https://github.com/elvisyjlin/AttGAN-PyTorch#to-test-with-your-custom-images-supports-testpy-test_multipy-test_slidepy).

**Comment by Jnasic** [comment]
Thanks, I will look at it.
Too many thanks.
